### PR TITLE
Refactor/UI cleanup

### PR DESCRIPTION
**Pull request by mattyg**
Major refactoring of ui to reduce complexity, simplify setup of holochain client & profiles store, prepare for replacing quasar ui library if desired. Other minor improvements to user experiance

## Refactoring
- rename all components to TitleCase, modify vite build settings to treat any dash-case components as custom elements
- remove `ui-common-library` entirely, instead use `@holo-host/web-sdk` directly for holo backend (temporarily on fork waiting for bug fix https://github.com/Holo-Host/web-sdk/pull/95)
- simplify architecture for initializing hc client, profiles store
- eliminate inconsistent bugs related to profiles, hc client and app initialization
    - remove pinia store as abstraction between client and usage of client
    - simplify and centralize logic
    - add 'ping' zome and ui indicator displaying if cells are responsive yet
- extract search bar into its own component
- remove layer of abstraction between callZome and components, call zomes directly within components
- refactor props onPublishMew and onToggleLickMew to events
- replace usage of quasar plugin-based dialogs with standard QDialog and v-model
- fix issue in dev environment where one agent's front-end would be unresponsive with a bunch of console errors

## UX changes
- bump prefix index to 0.4.0 to fix minor bug
- remove delays for showing tooltips
- animate show/hide of ProfilePopup to match tooltip animation
- on hovering timestamp, show absolute time in tooltip

## Todo
- [x] bump holochain client to latest
- [x] bump profiles to latest
- [x] update build setup to support new requirements from latest hc client (waiting on PR fix to lit-svelte-stores to resolve build issue https://github.com/guillemcordoba/lit-svelte-stores/pull/3)
- [x] update all ui dependencies
- [x] make material icons a dependency to serve locally
- [x] add ui indicator of "cell launching" (currently first zome call hangs for ~10s while waiting for cell to launch without explanation)
- [ ] migrate back to official `@holo-host/web-sdk` once PR is merged (https://github.com/Holo-Host/web-sdk/pull/95)
- [ ] setup dependabot in github for managing dependency updates

### CODE REVIEW

File: ui/src/components/LinkProfilePopup.vue [hunk 1]
```diff
@@ -35,27 +37,20 @@ defineProps<{
   agentPubKey: Uint8Array;
 }>();
 
-const PROFILE_SHOW_HIDE_DELAY = 400; // in ms
```
Comment (jost-s) [inline ui/src/components/LinkProfilePopup.vue:38]: Nice that the popup doesn't jump around anymore. I had opened an issue for that a long time ago https://github.com/quasarframework/quasar/issues/13517
It's never been looked at =(

Could be that we want a delay on hover, but I'd say for the moment let's try without it.

**Review by jost-s — APPROVED** [review]
I realize that the "delete mews" PR was based on this one, so I've implicitly reviewed this one already. I was surprised to see so many "side effects" in the other PR, now I know why =)

As always I'm very happy with the refactorings!

**Comment by jost-s** [comment]
Did you coordinate with Robbie or someone from Holo what's the recommended approach for hybrid applications? It was the ui-common-library back in the day, I don't know what it is now.

**Comment by mattyg** [comment]
> Did you coordinate with Robbie or someone from Holo what's the recommended approach for hybrid applications? It was the ui-common-library back in the day, I don't know what it is now.

Yeah I talked with Robbie. Ui-common-library is abandoned. The @holo-host/web-sdk gives you a class WebSdkApi that's just an extension of AppAgentClient.

**Comment by jost-s** [comment]
Good stuff

**Pull request merged**